### PR TITLE
feat: dedicated toggle block component (#531)

### PR DESCRIPTION
**Pull request by iamnamananand996**
will close this issue: https://github.com/instill-ai/instill.tech/issues/531

Because

- Right now we are using the regular markdown toggle syntax and then cover custom style on it to implement toggle block. Here is the implementation
```
<details>
  <summary></summary>
</details>
```
This commit

- dedicated toggle block (with component name ToggleBlock) with MDX and refactor all the toggle block in docs to adapt this new MDX component.

### CODE REVIEW

File: src/components/ui/MdxToggleBlock.tsx
[the remark this answers is not in the document]
Comment (EiffelFly) [reply src/components/ui/MdxToggleBlock.tsx:1]: @iamnamananand996 Sorry, I make a mistake here, I think we can use ToggleBlock here

**Comment by iamnamananand996** [comment]
@EiffelFly , I have pushed the indentation fixes for ToggleBlock.

File: src/components/ui/MdxToggleBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import * as classNames from "classnames";
+
+export type MdxToggleBlockProps = {
+  children: ReactElement;
+  styles: string;
+};
+
+export const MdxToggleBlock = ({ children, styles }: MdxToggleBlockProps) => {
+  return (
+    <div className={classNames.default("instill-scoped-info-block", styles)}>
```
Comment (EiffelFly) [inline src/components/ui/MdxToggleBlock.tsx:11]: @iamnamananand996 Please help us replace `instill-scoped-info-block` with `instill-scoped-toggle-block`.

**Review by EiffelFly — COMMENTED** [review]
@iamnamananand996 Please help us through replace this naming

**Comment by iamnamananand996** [comment]
updated the above suggested changes

cc - @EiffelFly

**Review by EiffelFly — APPROVED** [review]
LGTM!! @iamnamananand996 Thanks for your great contribution

**Pull request merged**